main/zoe/guide/working_with_ZCFSeats.md

```diff
@@ -0,0 +1,229 @@
+offer by the contract. 
+
+- `UserSeats` represent offers external to Zoe and the contract.. They are
+the representation of the offer on the user side and let a contract instance
+see what the user has escrowed and ensure when the user gets a payout, 
+they get the right amount. 
+
+For example, in an auction, the person putting up the item has both a `ZCFSeat`,
+and a `UserSeat` as do each of the bidders. When the auction ends, the contract instance uses these `ZCFSeats` to reallocate the assets between the seller and the  winning bidder. The bidders use their `UserSeats` to check in on what's happening 
+with the auction contract instance. When the auction ends, `UserSeat` allocations
+are changed to reflect the resulting asset allocation, and payouts go to the `UserSeats`.
+
+## ZCFSeats
+
+A `ZCFSeat` can synchronously query its
+associated offer's current state, such as its currently
+allocated asset amounts. It can also synchronously 
+manipulate the offer.
+
+`ZCFSeats` have a *staged allocation*.  This is the current allocation
+value used if and when the seat is *reallocated*  over. Reallocation occurs
+when an offer successfully exits and the assets involved are
+redistributed to the contract instance participants based on the
+instance's result. 
+
+A seat's staged allocation starts out as empty. An empty allocation
+means when the offer resolves, no assets are distributed via this
+seat. **tyg todo: Is "empty" the right word? The operation is "clear",
+but I think "empty" conveys better the effect at least** There are
+operations to add and subtract amounts from a seat's staged
+allocation as offers are made and rejected, as well as to clear it
+back to its initial empty state.  
+
+A `ZCFSeat` has methods that do the following:
+- Create and exit a seat.
+- Get the seat's 
+   - Offer proposal. 
+   - Amount allocated for one keyword. 
+   - Current allocation for all keywords. 
+   - Current staged allocation.
+- Check if a proposed allocation satisfies offer safety. 
+- Manipulate a seat's staged allocation:
+   - Check if the seat has a non-empty staged allocation.
+   - Add an amount to the staged allocation.
+   - Subtract an amount from the staged allocation.
+   - Clear the staged allocation.
+- Associate a notifier with the seat's allocation.
+
+Also, `zcf.reallocate()` is called over an offer's `ZCFseats` with
+non-empty staged allocations. 
+
+### Creating and Exiting a `ZCFSeat`
+
+#### Creating a `ZCFSeat`
+
+To create a new, empty, `ZCFSeat`, use `zcf.makeEmptySeatKit()`. It
+returns both a new `ZCFSeat` and a Promise for a `UserSeat`
+
+Zoe uses seats to represent offers, and has two seat facets (a
+particular view or API of an object; there may be multiple such APIs
+per object) a `ZCFSeat` and a `UserSeat`. 
+
+```js
+const { zcfSeat: mySeat } = zcf.makeEmptySeatKit();
+```
+#### Exiting a `ZCFSeat`
+
+When you have finished processing an offer, or it fails for some
```

Comment:
Instead of `you`, we should say something like, "the contract code has finished processing an offer" and "the contract code exits the seat"